prov/tcp/src/tcpx_ep.c

```diff
@@ -395,14 +395,25 @@ static void tcpx_ep_tx_rx_queues_release(struct tcpx_ep *ep)
 
 static int tcpx_ep_close(struct fid *fid)
 {
+	struct tcpx_eq *eq;
 	struct tcpx_ep *ep = container_of(fid, struct tcpx_ep,
 					  util_ep.ep_fid.fid);
 
+	eq = container_of(ep->util_ep.eq, struct tcpx_eq,
+			  util_eq);
+
 	tcpx_ep_tx_rx_queues_release(ep);
-	tcpx_cq_wait_ep_del(ep);
+
+	/* eq->lock protects from processing stale ep connection
+	   events*/
+	fastlock_acquire(&eq->lock);
```

Comment:
Hmm... this lock looks suspect.

Reply:
this lock is held while processing the events from eq wait obj. So by holding this lock here, i hope to avoid eq processing the events generated by this ep after it is removed from eq wait.